torch/optim/sgd.py

```diff
@@ -80,7 +80,7 @@ class SGD(Optimizer):
     """
 
     def __init__(self, params, lr=required, momentum=0, dampening=0,
-                 weight_decay=0, nesterov=False):
+                 weight_decay=0, nesterov=False, maximize=False):
```

Comment:
cc @jbschlosser  Do we want to make this kwarg only? For the other functions as well?

Reply:
From offline discussion, he agrees.
Can you make sure that this is a kwarg-only argument by doing: `nesterov=False, *, maximize=False):` please?

Reply:
Will do -- I prefer that too.

Reply:
I have made this change, however it can't be applied in `torch/distributed/optim/functional_sgd.py` as that is scripted and scripting does not support this:
```
  File "/home/oliver/code/playground/maximisation/torch/jit/frontend.py", line 337, in build_param_list
    raise NotSupportedError(ctx_range, _vararg_kwarg_err)
torch.jit.frontend.NotSupportedError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults:
  File "/home/oliver/code/playground/maximisation/torch/distributed/optim/functional_sgd.py", line 28
        _allow_empty_param_list: bool = False,
        *,
        maximize: bool = False,
                         ~~~~~ <--- HERE
    ):
```

This is an annoying issue to introduce -- would documentation be enough to cover for this?

Reply:
Upon further reflection, this is not an issue at all, since the `ZeroRedundancyOptimizer` constructor only accepts `**defaults` to pass to the optimizer constructor, so all args are key word only :D